### PR TITLE
Fix GrpcStatus for InvalidProtocolBufferException

### LINKED ISSUE #5379: grpc UNAVAILABLE status returned when calling grpc-json rpc with invalid Number (Long, Integer, ...)
When using `HttpJsonTranscoding` with grpc, if we call an rpc with an invalid value for a number, we get a grpc status UNAVAILABLE and no log, which makes the diagnostic very cumbersome.

Exemple:
```
service MyService {
    rpc TestArmeriaGrpcIntFailure(TestArmeriaGrpcIntFailureRequest) returns (TestArmeriaGrpcIntFailureResponse) {}
}

message TestArmeriaGrpcIntFailureRequest {
    int64 number = 1;
}

message TestArmeriaGrpcIntFailureResponse {

}
```

Create an armeria server with this proto file, enable HttpJsonTranscoding, and call 
```
curl -XPOST -H 'content-type: application/json; charset=utf-8; protocol=gRPC' http://localhost:8080'/MyService/TestArmeriaGrpcIntFailure' -d '{
  "number": 10.12
}'
```
We get:
```
< HTTP/1.1 503 Service Unavailable
< content-type: application/json; charset=utf-8
< grpc-status: 14
< content-length: 37
< server: Armeria/1.26.2
< date: Wed, 10 Jan 2024 15:21:11 GMT
<

{"code":14,"grpc-code":"UNAVAILABLE"}
```

Note the 10.12, which is a valid json Number, but can't be parsed to a `Long`.

I've tried to check the source code, and I see that a `Long.parseLong` is used, and it seems the NumberFormatException raised is catched here:
https://github.com/line/armeria/blob/30c9718d54c29d64ce1169b932618d43e0a37479/grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingService.java#L572

still I don't know why it results in a UNAVAILABLE status, but at least it would be nice to be able to log something in such case (or have a custom mapping exception handler).

### PR DESCRIPTION
**Pull request by jaeseung-bae**
Motivation:
- Closes #5379 

Modifications:

- Add corner case(`InvalidProtocolBufferException`) handling logic that returns `INVALID_ARGUMENT` to give more helpful description to clients.
- Modify `rpc EchoWrappers()` in transcoding.proto to accept http-post-binding in order to add integration test for type-mismatched json value
- Add unit-test, integration-test

Result:

- Closes #5379 
- Describe the consequences that a user will face after this PR is merged.
  - User gets `INVALID_ARGUMENT` instead of `UNAVAILABLE` for parsing exception when using `HttpJsonTranscoding`.

### CODE REVIEW

File: grpc/src/test/java/com/linecorp/armeria/internal/common/grpc/GrpcStatusTest.java [hunk 1]
```diff
@@ -58,4 +59,10 @@ void failFastExceptionToUnavailableCode() {
                              .getCode())
                 .isEqualTo(Status.Code.UNAVAILABLE);
     }
+
+    @Test
+    void invalidProtocolBufferExceptionToInvalidArgumentCode() {
+        assertThat(GrpcStatus.fromThrowable(new InvalidProtocolBufferException("Failed to parse message")).getCode())
+                .isEqualTo(Status.Code.INVALID_ARGUMENT);
+    }
```
Comment (ikhoon) [inline grpc/src/test/java/com/linecorp/armeria/internal/common/grpc/GrpcStatusTest.java:67]: Could we also add an integration test that sends invalid data described in the issue?

Comment (jaeseung-bae) [reply grpc/src/test/java/com/linecorp/armeria/internal/common/grpc/GrpcStatusTest.java:67]: Okay, I'll

**Review by minwoox — APPROVED** [review]
Looks great, thanks!

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

**Pull request merged**